src/common/libraries/MessageLib.sol

```diff
@@ -126,7 +129,9 @@ library MessageLib {
         uint8 kind = message.toUint8(0);
         require(kind <= uint8(type(MessageType).max), UnknownMessageType());
```

Comment:
This will still break with more than 32 messages because be cannot serialize the kind to `uint8` then anymore. I don't think there is a way around using `uint16` here.

Reply:
Why not? `uint8` can hold 256 different messages

Reply:
Sorry, logical mistake on my side. You are right.